### PR TITLE
Yaml support

### PR DESCRIPTION
**Pull request by mtyurt**
Hello,
I have completed the implementation of yaml module. Any feedback is welcome. 
There is one basic issue, the media type. I have typed the media type as `application/yaml`. I hope it's appropriate.
Also, we can discuss test outputs. For instance, in the root type test, the output is 

> ---
> - !\<bean>

>   a: 3
> - !\<bean>

>  a: 3

There are other test results we should look into, but not as much as this one.

### CODE REVIEW

File: yaml/pom.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      <version>${version.jackson.yaml}</version>
+    </dependency>
+    <!-- may also need JAXB annotation support -->
+    <dependency> 
+      <groupId>com.fasterxml.jackson.module</groupId>
+      <artifactId>jackson-module-jaxb-annotations</artifactId>
+      <version>${version.jackson.jaxb}</version>
+    </dependency>
+    <!--  But Stax2 API must be included -->
+    <dependency>
```
Comment (cowtowncoder) [inline yaml/pom.xml]: Woodstox and stax2 only needed by XML module, so these can be removed. JAXB annotation support does not need them.

File: yaml/src/main/java/com/fasterxml/jackson/jaxrs/yaml/JacksonJaxbYAMLProvider.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * provider for JAX-RS and use as is: this will use both Jackson and
+ * JAXB annotations (with Jackson annotations having priority).
+ *<p>
+ * Note: class annotations are duplicated from super class, since it
+ * is not clear whether JAX-RS implementations are required to
+ * check settings of super-classes. It is important to keep annotations
+ * in sync if changed.
+ */
+@Provider
+@Consumes({"text/yaml", "application/yaml"})
```
Comment (cowtowncoder) [inline yaml/src/main/java/com/fasterxml/jackson/jaxrs/yaml/JacksonJaxbYAMLProvider.java]: I assume these are correct, but just wanted to ask if you checked IETF/RFC suggestions for YAML?
Not a big deal as these can be changed easily.

File: yaml/src/main/java/com/fasterxml/jackson/jaxrs/yaml/JsonMappingExceptionMapper.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.fasterxml.jackson.jaxrs.yaml;
+
+import javax.ws.rs.ext.Provider;
+
+/**
+ * @deprecated Since 2.2 use {@link com.fasterxml.jackson.jaxrs.base.JsonMappingExceptionMapper} instead.
+ */
+@Deprecated
```
Comment (cowtowncoder) [inline yaml/src/main/java/com/fasterxml/jackson/jaxrs/yaml/JsonMappingExceptionMapper.java]: Let's not add this -- deprecated for existing modules, so better not to include for new ones, as they can use the shared standard one.

File: yaml/src/main/java/com/fasterxml/jackson/jaxrs/yaml/JsonParseExceptionMapper.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.fasterxml.jackson.jaxrs.yaml;
+
+import javax.ws.rs.ext.Provider;
+
+/**
+ * @deprecated Use {@link com.fasterxml.jackson.jaxrs.base.JsonParseExceptionMapper} instead.
+ */
+@Deprecated
```
Comment (cowtowncoder) [inline yaml/src/main/java/com/fasterxml/jackson/jaxrs/yaml/JsonParseExceptionMapper.java]: And same here, deprecated, no need to add.

File: yaml/src/main/java/com/fasterxml/jackson/jaxrs/yaml/annotation/JacksonFeatures.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Annotation that can be used enable and/or disable various
+ * features for <code>ObjectReader</code>s and <code>ObjectWriter</code>s.
+ * 
+ * @deprecated Since 2.2, use shared {@link com.fasterxml.jackson.jaxrs.annotation.JacksonFeatures} instead
+ */
+@Target({ElementType.ANNOTATION_TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+@com.fasterxml.jackson.annotation.JacksonAnnotation
+@Deprecated // since 2.2
```
Comment (cowtowncoder) [inline yaml/src/main/java/com/fasterxml/jackson/jaxrs/yaml/annotation/JacksonFeatures.java]: And this is deprecated, no need to add.

**Comment by cowtowncoder** [comment]
Looks good, just need to trim couple of things. I can go and investigate YAML mime type just to confirm.

Good question on root serialization, configurability.

File: yaml/src/main/java/com/fasterxml/jackson/jaxrs/yaml/JacksonJaxbYAMLProvider.java (hunk 2 above)
Comment (mtyurt) [inline yaml/src/main/java/com/fasterxml/jackson/jaxrs/yaml/JacksonJaxbYAMLProvider.java]: Well, apparently there is not a standard for this although there is convention. I mostly referred to [this](http://stackoverflow.com/questions/332129/yaml-mime-type) question and answers.

**Comment by cowtowncoder** [comment]
Ok, yes makes sense. I did not realize that 'text/' variants are theoretically to consider input as ISO-8859-1, vs `application/`.

**Pull request closed without merging**